project-base/storefront/components/Pages/Order/PaymentConfirmation/helpers.ts

```diff
@@ -30,6 +32,7 @@ export const useUpdatePaymentStatus = (orderUuid: string, orderPaymentStatusPage
         ) {
             return;
         }
+        fetchContent(updatePaymentStatusActionResult.data.UpdatePaymentStatus.isPaid);
```

Comment:
Why can't we fetch data in with genereted hooks in each Payment `PaymentSuccess` and `PaymentFail` componenent? I guess it is made as generic here because of loading state, right? Why not handle loading state properly in those components?

Reply:
I did it this way, because we need to control one skeleton while loading the status and then a second one while loading the content. This way we can handle it in one place. There are other approaches I have considered, such as:

- show a skeleton for status and a different one for content (they would look the same, but would be controlled from a different component)
- have one loading state which would be controlled from two components (by callbacks or something)

Do you consider any of them to be better?

Reply:
I personally would vote for third option which is handling those two loading state separately. Since it actually is two different states and I don't like this `content` prop. Actual solution is causing complexity and is more difficult to debug. 

The loading transition shouldn't be even visible in my understanding. Since after load the page fetching for content should start immediately with the component render. And big benefit is that we can use generated urql hook and don't care about handling those states at all.

Reply:
Nice, me personally I would call this as nice improvement :)